File: python3/Two Pointers/is_palindrome_valid.py
def is_palindrome_valid(s: str) -> bool:    
    left, right = 0, len(s) - 1
    while left < right:   
        # Skip non-alphanumeric characters from the left.     
        while left < right and not s[left].isalnum():
            left += 1
        # Skip non-alphanumeric characters from the right.
        while left < right and not s[right].isalnum():
            right -= 1
        # If the characters at the left and right pointers don't
        # match, the string is not a palindrome.
        if s[left] != s[right]:
            return False 
        left += 1
        right -= 1
    return True

File: go/Two Pointers/is_palindrome_valid.go
func isAlnum(r rune) bool {
	return unicode.IsLetter(r) || unicode.IsDigit(r)
}

func isPalindromeValid(input string) bool {
	s := []rune(input)

	left, right := 0, len(s)-1

	for left < right {
		// Skip non-alphanumeric characters from the left.
		for left < right && !isAlnum(s[left]) {
			left++
		}
		// Skip non-alphanumeric characters from the right.
		for left < right && !isAlnum(s[right]) {
			right--
		}
		// If the characters at the left and right pointers don't
		// match, the string is not a palindrome.
		if s[left] != s[right] {
			return false
		}

		left++
		right--
	}

	return true
}
